### ISSUE TITLE
Setup with React Native, the latest version of the problem

### ISSUE THREAD

**Issue by chinavane** [GeekyAnts/vue-native-core#84]
```
1.react-native init vnStart

2.cd vnStart

3.
npm install vue-native-core vue-native-helper --save
npm install vue-native-scripts --save-dev

4.vueTransformerPlugin.js
var semver = require('semver');
var vueNaiveScripts = require("vue-native-scripts");
var reactNativeVersionString = require("react-native/package.json").version;
var reactNativeMinorVersion = semver(reactNativeVersionString).minor;

if (reactNativeMinorVersion >= 56) {
  upstreamTransformer = require("metro/src/reactNativeTransformer");
} else if (reactNativeMinorVersion >= 52) {
  upstreamTransformer = require("metro/src/transformer");
} else if (reactNativeMinorVersion >= 47) {
  upstreamTransformer = require("metro-bundler/src/transformer");
} else if (reactNativeMinorVersion === 46) {
  upstreamTransformer = require("metro-bundler/build/transformer");
} else {
  // handle RN <= 0.45
  var oldUpstreamTransformer = require("react-native/packager/transformer");
  upstreamTransformer = {
    transform({ src, filename, options }) {
      return oldUpstreamTransformer.transform(src, filename, options);
    }
  };
}
var vueExtensions = ["vue"]; // <-- Add other extensions if needed.

module.exports.transform = function({ src, filename, options }) {
  if (vueExtensions.some(ext => filename.endsWith("." + ext))) {
    return vueNaiveScripts.transform({ src, filename, options });
  }
  return upstreamTransformer.transform({ src, filename, options });
};


5.rn-cli.config.js
module.exports = {
  getTransformModulePath() {
    return require.resolve("./vueTransformerPlugin.js");
  },
  getSourceExts() {
    return ["vue"];
  }
};


6.rename App.js to App.vue
<template>
  <view class="container">
    <text class="text-color-primary">My Vue Native App</text>
    </view>
</template>
<style>
.container {
  background-color: white;
  align-items: center;
  justify-content: center;
  flex: 1;
}
.text-color-primary {
  color: blue;
}
</style>

7.package.json,react-native is the lasted version.
{
  "name": "vnStart",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "react": "16.5.0",
    "react-native": "0.57.0",
    "vue-native-core": "0.0.8",
    "vue-native-helper": "0.0.9"
  },
  "devDependencies": {
    "babel-jest": "23.6.0",
    "jest": "23.6.0",
    "metro-react-native-babel-preset": "0.45.2",
    "react-test-renderer": "16.5.0",
    "vue-native-scripts": "0.0.14"
  },
  "jest": {
    "preset": "react-native"
  }
}

8. .babelrc
{
  "presets": ["module:metro-react-native-babel-preset"]
}
```

![snip20180914_5](https://user-images.githubusercontent.com/8588836/45522606-8a9a3580-b7f6-11e8-99c3-cad48dc4cabd.png)

![snip20180914_2](https://user-images.githubusercontent.com/8588836/45522530-27a89e80-b7f6-11e8-8c0d-7c072c05972d.png)

![snip20180914_1](https://user-images.githubusercontent.com/8588836/45522517-12337480-b7f6-11e8-99ef-99ae44b5aa13.png)

**Comment by chinavane** [comment]
![snip20180914_3](https://user-images.githubusercontent.com/8588836/45522562-545cb600-b7f6-11e8-8f3d-4c0c3103b13c.png)
![snip20180914_4](https://user-images.githubusercontent.com/8588836/45522568-5d4d8780-b7f6-11e8-9c66-b54ecb0f5c5b.png)


I have to lower the development version to run the vue project normally.

**Comment by neeraj-singh47** [comment]
@chinavane Yes currently we are only supporting react-native <= 0.55.4

**Comment by tonismi** [comment]
any plans for near future to support react native  0.57 with vue? maybe some suggestion how to configure it using last typecript setup for 0.57?

**Comment by neeraj-singh47** [comment]
@tonismi Yes, this is in roadmap. We'll add support for React Native > 55 in future

**Comment by tonismi** [comment]
Have you defined real guides of roadmap? Some doubts to spent or not the time with this tool , because there are no movements for long time, also react-vue projects from SmallComfort and GeekyAnts were abandoned 7-12 months ago, in the state of very early versions 0.0.x

**Comment by neeraj-singh47** [comment]
@tonismi We haven't abandoned vue-native project. We've added support render-prop and render-prop-fn so that you don't have to use jsx. We also included support for v-model and v-on. Also if you read expo's blog https://blog.expo.io/expo-sdk-30-0-0-is-now-available-e64d8b1db2a7. They are not supporting 0.56 or 0.57 as well. We need to support both the frameworks. We'll surely add supports for 0.56 and 0.57 in future
